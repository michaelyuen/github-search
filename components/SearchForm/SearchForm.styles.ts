import styled from "styled-components";
import { StyledInput } from "../Input";

export const SearchFormContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 4em;

  ${StyledInput} {
    flex-grow: 1;
  }

  @media (min-width: 768px) {
    h3 {
      margin-right: 6em;
    }
  }
`;
