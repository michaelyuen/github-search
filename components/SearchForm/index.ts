export * from "./SearchForm";
export * from "./styles";
