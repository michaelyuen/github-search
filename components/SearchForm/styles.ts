import styled from "styled-components";
import { StyledInput } from "../Input";
import { Form } from "../Form";
import { breakpoints } from "../../styles/breakpoints";
import { ErrorContainer } from "../Error";

export const SearchFormContainer = styled(Form)`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 4em;

  h2 {
    flex-shrink: 0;
  }

  div {
    display: flex;
    flex-grow: 1;
    flex-wrap: wrap;
    position: relative;

    ${StyledInput} {
      flex-grow: 1;
    }

    ${ErrorContainer} {
      margin-bottom: 0;
    }
  }

  @media (min-width: ${breakpoints.md}) {
    flex-wrap: nowrap;

    h2 {
      margin-right: 6em;
    }

    ${ErrorContainer} {
      left: 0;
      margin: 0;
      position: absolute;
      top: 100%;
    }
  }
`;
