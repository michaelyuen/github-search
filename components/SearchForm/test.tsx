import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import { SearchForm } from "./SearchForm";

describe("SearchForm", () => {
  test("renders", () => {
    render(
      <SearchForm
        aria-label="mock aria label"
        onChange={jest.fn()}
        query=""
        title="GitHub Search Form"
      />
    );
    expect(screen.getByLabelText("mock aria label")).toBeInTheDocument();
  });

  test("renders error", () => {
    render(
      <SearchForm
        aria-label="mock aria label"
        error="Something went wrong. Please try again."
        onChange={jest.fn()}
        query=""
        title="GitHub Search Form"
      />
    );
    expect(screen.getByLabelText("mock aria label")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <SearchForm
        aria-label="mock aria label"
        onChange={jest.fn()}
        query=""
        title="GitHub Search Form"
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("has no a11y violations", async () => {
    const { container } = render(
      <SearchForm
        aria-label="mock aria label"
        onChange={jest.fn()}
        query=""
        title="GitHub Search Form"
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});
