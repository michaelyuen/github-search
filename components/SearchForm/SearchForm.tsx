import { SearchFormContainer } from "./styles";
import { Input, InputProps } from "../Input";
import { FormProps } from "../Form";
import { Error, ErrorProps } from "../Error";

export interface SearchFormProps {
  error?: ErrorProps["message"] | null;
  name?: FormProps["name"];
  onChange?: InputProps["onChange"];
  onKeyUp?: InputProps["onKeyUp"];
  onSubmit?: FormProps["onSubmit"];
  query: string;
  title: string;
}

export const SearchForm: React.FC<SearchFormProps> = ({
  error,
  onChange,
  onKeyUp,
  onSubmit,
  query,
  title,
  ...rest
}) => {
  return (
    <SearchFormContainer
      aria-label="GitHub Search Form"
      name="GitHub Search Form"
      onSubmit={onSubmit}
      {...rest}
    >
      <h2>{title}</h2>
      <div>
        <Input
          aria-label="Search using GitHub's advanced search syntax. Only user and stars qualifiers are supported."
          onChange={onChange}
          onKeyUp={onKeyUp}
          placeholder="Try user or stars qualifiers. e.g., user:atom stars:10..50"
          value={query}
        />
        {error && <Error message={error} />}
      </div>
    </SearchFormContainer>
  );
};
