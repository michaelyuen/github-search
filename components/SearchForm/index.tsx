import { SearchFormContainer } from "./SearchForm.styles";
import { Input } from "../Input";

export interface SearchFormProps {
  title: string;
}

export const SearchForm: React.FC<SearchFormProps> = ({ title }) => {
  return (
    <SearchFormContainer>
      <h3>{title}</h3>
      <Input />
    </SearchFormContainer>
  );
};
export { SearchFormContainer };
