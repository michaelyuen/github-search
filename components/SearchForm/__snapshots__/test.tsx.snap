// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches snapshot 1`] = `
<div>
  <form
    aria-label="mock aria label"
    class="sc-gswNZR sc-hLBbgP iPeGWR"
    name="GitHub Search Form"
  >
    <h2>
      GitHub Search Form
    </h2>
    <div>
      <input
        aria-label="Search using GitHub's advanced search syntax. Only user and stars qualifiers are supported."
        class="sc-bcXHqe iggXoH"
        data-testid="input"
        placeholder="Try user or stars qualifiers. e.g., user:atom stars:10..50"
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
